java/vector/src/main/java/org/apache/arrow/vector/validate/ValidateVectorVisitor.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.validate;
+
+import java.util.List;
+
+import org.apache.arrow.vector.BaseFixedWidthVector;
+import org.apache.arrow.vector.BaseVariableWidthVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.NullVector;
+import org.apache.arrow.vector.compare.VectorVisitor;
+import org.apache.arrow.vector.complex.DenseUnionVector;
+import org.apache.arrow.vector.complex.FixedSizeListVector;
+import org.apache.arrow.vector.complex.ListVector;
+import org.apache.arrow.vector.complex.NonNullableStructVector;
+import org.apache.arrow.vector.complex.UnionVector;
+import org.apache.arrow.vector.types.pojo.Field;
+
+/**
+ * visitor to validate vector (not include data).
+ */
+public class ValidateVectorVisitor implements VectorVisitor<Void, Void> {
+
+  @Override
+  public Void visit(BaseFixedWidthVector vector, Void value) {
+    if (vector.getValueCount() > 0) {
```

Comment:
do we need to check the validity buffer?

Reply:
We already check all buffers capacity in ValueVectorUtility

Reply:
I think we need to clarify how this visitor is going to be used. IMO, there are at least two possibilities:
1. The visitor is designed to be an independent feature.
2. The visitor is always used by ValueVectorUtility#validate.

Method 1 provides more flexibility and for method 2, we need to make it package private?
Method 1 provides

Reply:
Thanks, I prefer method1 and add some docs.